android/src/main/java/com/dicetechnology/rnpayments/RNPaymentsAmazonModule.java

```diff
@@ -0,0 +1,285 @@
+                        }
+                    } catch (Exception e) {
+                        rejectPromises(PromiseConstants.GET_PURCHASE_UPDATES, "BILLING_RESPONSE_JSON_PARSE_ERROR", e.getMessage(), e);
+                    }
+                    break;
+                case FAILED:
+                    rejectPromises(PromiseConstants.GET_PURCHASE_UPDATES, "FAILED IN onPurchaseUpdatesResponse: ", null, null);
+                    break;
+                case NOT_SUPPORTED:
+                    rejectPromises(PromiseConstants.GET_PURCHASE_UPDATES, "onPurchaseUpdatesResponse NOT_SUPPORTED", "Should retry request", null);
+                    break;
+            }
+        }
+
+        public void onUserDataResponse(UserDataResponse userDataResponse) {
+            final UserDataResponse.RequestStatus status = userDataResponse.getRequestStatus();
+            switch (status) {
+                case SUCCESSFUL:
+                    try {
+                        UserData userData = userDataResponse.getUserData();
+
+                        WritableMap map = Arguments.createMap();
+                        map.putString("userId", userData.getUserId());
+                        map.putString("marketplace", userData.getMarketplace());
+
+                        resolvePromises(PromiseConstants.GET_USER_DATA, map);
+                    } catch (Exception e) {
+                        // TODO: If above works w/o error may not need the below catch block
+                        rejectPromises(PromiseConstants.GET_USER_DATA, "USER_DATA_RESPONSE_JSON_PARSE_ERROR", e.getMessage(), e);
+                    }
+                    break;
+                case FAILED:
+                    rejectPromises(PromiseConstants.GET_USER_DATA, "FAILED IN onUserDataResponse: ", null, null);
+                    break;
+                case NOT_SUPPORTED:
+                    rejectPromises(PromiseConstants.GET_PURCHASE_UPDATES, "onUserDataResponse NOT_SUPPORTED", "Should retry request", null);
+                    break;
+            }
+        }
+    };
+
+    public RNPaymentsAmazonModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+        registerListener(reactContext);
+    }
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "RNPaymentsAmazonModule";
+    }
+
+    @Nullable
+    @Override
+    public Map<String, Object> getConstants() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("APP_STORE_AVAILABLE", Utils.getAppStore(getReactApplicationContext()) == AppStore.AMAZON);
+        return map;
+    }
+
+    // Amazon supports skus up to 100 - needs pagination if more is to be loaded
+    @ReactMethod
+    public void loadProducts(ReadableArray skus, Promise promise) {
+        putPromise(PromiseConstants.LOAD_PRODUCTS, promise);
+        Set<String> set = new HashSet<>();
+        for (int i = 0; i < skus.size(); i++)
```

Comment:
for better readability please always use curly brackets, even in a single line "for" statements.